### DISCUSSION TITLE
non basic Proxy Authorization requests blocked by http_proxy raising error

### DISCUSSION THREAD

**Discussion by pettzilla1** [encode/httpcore#926] [General]
https://github.com/encode/httpcore/blob/c96fdf7d31c3ab983f4eec4ef7a25cd81edc7ac9/httpcore/_async/http_proxy.py#L293
The line filtering out status codes above 299 is overly restrictive as it filters out 407 and 401 requests which are requests for additional information for authorization,
I have tested a fix for this myself by adding to the line prior 

`if connect_response.status in(407,401):
    return( connect_response)`
this will allow the request for proxy authorization to be returned and followed up upon appropriately by the user.

I'm happy to write a pull request to action this update if there is capacity to review

**Comment by lovelydinosaur** [comment]
Raising a `ProxyError` in response to a 407 is a reasonable behavior here.

Related... could contributors please start with discussions first...

<img width="612" alt="image" src="https://github.com/encode/httpcore/assets/647359/45bad78d-0c9b-496f-9e8c-a70ab83be6a6">

...that allows us as a team to reserve "issues" for work that we've determined to be actionable & desired.

**Reply by pettzilla1** [reply]
But 407 is that the client has been unable to authenticate with the proxy, 
often elements of the response message are needed to authenticate with the proxy, raising a proxy error doesn’t allow the user to receive the elements of the response needed to authenticate. 
This is quite common with call and response authentication as the responses headers need to be returned so a connection can be made see https://www.xcitium.com/ntlm-explained/


I would argue raising an error before checking to see if any verification parameters have been passed is invalid as the error isn’t passing back all the relevant data
